pex/interpreter.py

```diff
@@ -979,9 +998,6 @@ def _spawn_from_binary(cls, binary):
         cached_interpreter = cls._PYTHON_INTERPRETER_BY_NORMALIZED_PATH.get(canonicalized_binary)
         if cached_interpreter is not None:
             return SpawnedJob.completed(cached_interpreter)
-        if canonicalized_binary == cls.canonicalize_path(sys.executable):
-            current_interpreter = cls(PythonIdentity.get())
```

Comment:
This was the bug. Here `PythonIdentity.get()` slurps up the ambient `sys.path` which may have been mucked with.